src/binarySearch.js

```diff
@@ -0,0 +1,30 @@
+export default function binarySearch( arr, query){
+  if( arr == null ){
+    return 'Cannot search null'
+  }
+  if( arr.constructor !== Array ){
+    return 'Please use Arrays'
+  }
+
+  const splitSearch = (start, end) => {
+    var midpoint = Math.round( (end-start)/2 + start)
+
+    if( query === arr[midpoint]){
+      return midpoint
+    }
+
+    if( query > arr[midpoint] ){
+      if( start === end ){
```

Comment:
We do need it in both places for different circumstances.  In one case it's needed to catch items larger than any in the array, and in another it's needed to catch items smaller than any in the array.  If we remove either case, then we get infinite loops in our test somewhere.

Perhaps we could rewrite the sort so that it's only needed in one location?